bitwuzla/src/bitwuzla_solver.cpp

```diff
@@ -0,0 +1,694 @@
+  }
+}
+
+Result BzlaSolver::check_sat_assuming(const TermVec & assumptions)
+{
+  return check_sat_assuming_internal(assumptions.begin(), assumptions.end());
+}
+
+Result BzlaSolver::check_sat_assuming_list(const TermList & assumptions)
+{
+  return check_sat_assuming_internal(assumptions.begin(), assumptions.end());
+}
+
+Result BzlaSolver::check_sat_assuming_set(
+    const UnorderedTermSet & assumptions)
+{
+  return check_sat_assuming_internal(assumptions.begin(), assumptions.end());
+}
+
+void BzlaSolver::push(uint64_t num) { bitwuzla_push(bzla, num); }
+
+void BzlaSolver::pop(uint64_t num) { bitwuzla_pop(bzla, num); }
+
+Term BzlaSolver::get_value(const Term & t) const
+{
+  shared_ptr<BzlaTerm> bterm = static_pointer_cast<BzlaTerm>(t);
+  return make_shared<BzlaTerm>(bitwuzla_get_value(bzla, bterm->term));
+}
+
+UnorderedTermMap BzlaSolver::get_array_values(const Term & arr,
+                                              Term & out_const_base) const
+{
+  throw NotImplementedException(
+      "Bitwuzla backend doesn't support get_array_values yet");
+}
+
+void BzlaSolver::get_unsat_assumptions(UnorderedTermSet & out)
+{
+  size_t size;
+  BitwuzlaTerm ** bcore = bitwuzla_get_unsat_assumptions(bzla, &size);
+  for (size_t i = 0; i < size; ++i)
+  {
+    assert(*bcore);
+    out.insert(make_shared<BzlaTerm>(*bcore));
+    bcore++;
+  }
+}
+
+Sort BzlaSolver::make_sort(const string name, uint64_t arity) const
+{
+  throw NotImplementedException(
+      "Bitwuzla backend does not support uninterpreted sorts");
+}
+
+Sort BzlaSolver::make_sort(SortKind sk) const
+{
+  if (sk == BOOL)
+  {
+    return make_shared<BzlaSort>(bitwuzla_mk_bool_sort(bzla));
+  }
+  else
+  {
+    throw NotImplementedException("Bitwuzla does not support sort "
+                                  + to_string(sk));
+  }
+}
+
+Sort BzlaSolver::make_sort(SortKind sk, uint64_t size) const
+{
+  if (sk == BV)
+  {
+    return make_shared<BzlaSort>(bitwuzla_mk_bv_sort(bzla, size));
+  }
+  else
+  {
+    std::string msg("Can't create sort from sort kind ");
+    msg += to_string(sk);
+    msg += " with int argument.";
+    throw IncorrectUsageException(msg);
+  }
+}
+
+Sort BzlaSolver::make_sort(SortKind sk, const Sort & sort1) const
+{
+  throw IncorrectUsageException(
+      "Bitwuzla has no sort that takes a single sort argument.");
+}
+
+Sort BzlaSolver::make_sort(SortKind sk,
+                           const Sort & sort1,
+                           const Sort & sort2) const
+{
+  shared_ptr<BzlaSort> bsort1 = static_pointer_cast<BzlaSort>(sort1);
+  shared_ptr<BzlaSort> bsort2 = static_pointer_cast<BzlaSort>(sort2);
+
+  if (sk == ARRAY)
+  {
+    return make_shared<BzlaSort>(
+        bitwuzla_mk_array_sort(bzla, bsort1->sort, bsort2->sort));
+  }
+  else if (sk == FUNCTION)
+  {
+    vector<BitwuzlaSort *> domain_sorts({ bsort1->sort });
+    return make_shared<BzlaSort>(bitwuzla_mk_fun_sort(
+        bzla, domain_sorts.size(), &domain_sorts[0], bsort2->sort));
```

Comment:
Nicer: 
```suggestion
        bzla, domain_sorts.size(), domain.data(), bsort2->sort));
```

Reply:
Right, thanks! P.s. I have actually been using that since you told me about it and it is much nicer -- I never knew! This was just old.